pyro/contrib/gp/models/gpr.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import absolute_import, division, print_function
+
+import torch
+from torch.autograd import Variable
+import torch.nn as nn
+
+import pyro
+import pyro.distributions as dist
+
+
+class GPRegression(nn.Module):
+    """
+    Gaussian Process regression module.
+    """
+    def __init__(self, X, y, kernel, noise=torch.ones(1), priors={}):
+        super(GPRegression, self).__init__()
+        self.X = X
+        self.y = y
+        self.input_dim = X.size(0)
+        self.kernel = kernel
+        # TODO: define noise as a nn.Module, so we can train/set prior to it
+        self.noise = Variable(noise.type_as(X.data))
+        self.priors = priors
+
+    def model(self):
+        kernel_fn = pyro.random_module(self.kernel.name, self.kernel, self.priors)
+        kernel = kernel_fn()
+        K = kernel.K(self.X) + self.noise.repeat(self.input_dim).diag()
+        zero_loc = Variable(torch.zeros(self.input_dim).type_as(K.data))
```

Comment:
I think to get correct GPU device placement you'll need to
```py
zero_loc = K.new([0]).expand(self.input_dim)
```

Reply:
I see no problem with using `type_as` method (in pytorch ver 3). However, I just have 1 GPU to test. Did you mean that this will fail for multi-GPU?

Reply:
Yes, `K.new(...)` ensures that the result is on the same GPU as `K`, whereas I believe `.type_as()` only ensures that the result is on some GPU. I only learned this recently, and we still have a few bugs in Pyro relating to GPU placement.

Reply:
Now I try to use `.new()` whenever I create a new tensor.

Reply:
Aha, thank you a lot for the tip! Will change soon.